src/server/scripts/Spells/spell_dh.cpp

```diff
@@ -909,6 +910,43 @@ class areatrigger_dh_generic_sigil : public AreaTriggerEntityScript
     }
 };
 
+// 388116 - Shattered Destiny
+class spell_dh_shattered_destiny : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DH_METAMORPHOSIS_TRANSFORM });
+    }
+
+    void HandleProc(ProcEventInfo const& eventInfo)
+    {
+        Spell const* procSpell = eventInfo.GetProcSpell();
+        if (!procSpell)
+            return;
+
+        Aura* metamorphosis = GetTarget()->GetAura(SPELL_DH_METAMORPHOSIS_TRANSFORM);
+        if (!metamorphosis)
+            return;
+
+        int32 _furyAmount = GetEffect(EFFECT_1)->GetAmount();
```

Comment:
accessing EFFECT_1 without validation at OnProc hook, EFFECT_0 is accessed below, but I think that one is alright because it's assumed that 1 effect (EFFECT_0) exists